Replacing get_posts with wc_get_orders

wjrosa opened the pull request:
Fixes 2248-gh-woocommerce/woocommerce-subscriptions

## Description

<!--
Write a brief summary about this PR. 
- Why is this change needed? 
- What does this change do? 
- Were there other solutions you considered? 
- Why did you choose to pursue this solution? 
- Describe any trade-offs you might have had to make.
-->

This PR replaces the usage of the function `get_posts` with `wc_get_orders` to improve performance. To do that, I have updated all the parameters accordingly.

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos as appropriate.
-->

## How to test this PR

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
Use the testing instructions from the linked issue as a starting point.
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Code review. Check if the tests are still passing.

## Product impact
<!-- What products will this PR ship in? -->

- [x] Added changelog entry (or does not apply)
- [x] Will this PR affect WooCommerce Subscriptions? yes
- [x] Will this PR affect WooCommerce Payments? yes
- [ ] <!-- 🚨 Deprecations 🚨 --> Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)

wjrosa committed:
Replacing get_posts with wc_get_orders

wjrosa committed:
Changelog entry

mattallan reviewed (commented):
Hey @wjrosa I'm sorry to say but these functions will no longer be used and will be deprecated once we merge https://github.com/Automattic/woocommerce-subscriptions-core/pull/163 so I don't think we need to update them 😢 Sorry!

I think we can also ignore the `get_posts()` used in `wcs_get_subscription_id_from_key()` as it's an old helper function from back when a subscription object used to be an array of data stored in order meta (stored in meta `{$order_id}_{$product_id}` which we referred to as a subscriptions key) 🙈 

I guess it technically still works and it's not a deprecated helper 🤷 So maybe it's fine for us to update it? Although, just above this `get_posts()` is a direct `posts` table query so we'd need to update that as well.

I'll leave it up to you if you want to update it or close this PR entirely.

If we do decide to leave it, we should probably add an inline comment in case someone stumbles upon this `get_posts()` and has questions xD

wjrosa committed:
Merge branch 'trunk' into dev/replacing-get-posts-with-wc-get-orders

wjrosa committed:
Reverting change to wcs_get_subscription_id_from_key + adding comment above get_posts + fix lint issues

wjrosa committed:
Reverting unnecessary changes due deprecation

wjrosa committed:
Removing changelog entry

wjrosa commented:
> Hey @wjrosa I'm sorry to say but these functions will no longer be used and will be deprecated once we merge #163 so I don't think we need to update them 😢 Sorry!
> 
> I think we can also ignore the `get_posts()` used in `wcs_get_subscription_id_from_key()` as it's an old helper function from back when a subscription object used to be an array of data stored in order meta (stored in meta `{$order_id}_{$product_id}` which we referred to as a subscriptions key) 🙈
> 
> I guess it technically still works and it's not a deprecated helper 🤷 So maybe it's fine for us to update it? Although, just above this `get_posts()` is a direct `posts` table query so we'd need to update that as well.
> 
> I'll leave it up to you if you want to update it or close this PR entirely.
> 
> If we do decide to leave it, we should probably add an inline comment in case someone stumbles upon this `get_posts()` and has questions xD

No worries! Thanks for letting me know. I have just added the comment above the `get_posts` call instead and removed the changelog entry 👍

mattallan reviewed (approved):
Thanks @wjrosa for the latest changes! Approving this :)
